submodule.go

```diff
@@ -68,6 +80,55 @@ func addVendorSubmodule(importpath string) error {
 	return errors.New(strings.Join(errs, "\n"))
 }
 
+// NOTE: This list is copied from
+// https://github.com/golang/go/blob/10538a8f9e2e718a47633ac5a6e90415a2c3f5f1/src/cmd/go/vcs.go#L821-L861
+var wellKnownSites = []string{
+	"github.com/",
+	"bitbucket.org/",
+	"hub.jazz.net/git/",
+	"git.apache.org/",
+	"git.openstack.org/",
+}
+
+func getHttpsURLForImportPath(importpath string) (url string, err error) {
+	url = "https://" + importpath
+	for _, site := range wellKnownSites {
+		if strings.HasPrefix(importpath, site) {
+			return
```

Comment:
Please do not use naked return, I think it will be more clear if we will use it as `return url, nil`